### PR TITLE
Add decorators

### PR DESCRIPTION
**Pull request by Gopxfs**
# Summary 📒 
For this milestone, I:
- Added correct_name method to Person.
- Added Nameable class.
- Added class Decorator < Nameable.
- Added classes CapitalizeDecorator/TrimmerDecorator < Decorator.
- Updated class Person to < Nameable.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def correct_name
+    @person.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @person.correct_name[0, 10]
+  end
+end
```
Comment (uwadonat) [inline decorator.rb:22]: - [ ] As requested, let's trim the word if the name has a maximum of 10 character

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @Gopxfs,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Descriptive pull request message ✔️
- Good commit messages ✔️
- No linter errors ✔️
almost

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (Gopxfs) [reply decorator.rb:22]: Hey @uwadonat what does trim a word mean?

Comment (omar-muhamad) [reply decorator.rb:22]: > * [ ]  As requested, let's trim the word if the name has a maximum of 10 character

Hi @uwadonat, 
The trimmer decorator class is working just fine and I tested it see the image below.

image

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @Gopxfs,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: decorator.rb (hunk 1 above)
Comment (uwadonat) [reply decorator.rb:22]: Hi! Sorry for the delay. It was requested to check for the length of the word first to see if number of character exceed 10 before applying this method.

**Pull request merged**